main.go

```diff
@@ -39,9 +39,37 @@ type cliArgs struct {
 	KubeConfig string `arg:"-k, --kubeconfig, env" default:"" help:""`
 }
 
+type viewData struct {
+	Title   string
+	Message string
+}
+
 func OkHandler(w http.ResponseWriter, _ *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
+
+func CallbackSuccessHandler(w http.ResponseWriter, r *http.Request) {
+	http.ServeFile(w, r, "static/callback_success.html")
+}
+
+func CallbackErrorHandler(w http.ResponseWriter, r *http.Request) {
+	q := r.URL.Query()
+	errorMsg := q.Get("error")
+	errorDescription := q.Get("error_description")
+	data := viewData{
+		Title:   errorMsg,
+		Message: errorDescription,
+	}
+	tmpl, _ := template.ParseFiles("static/callback_error.html")
+
+	w.WriteHeader(http.StatusOK)
+	err := tmpl.Execute(w, data)
+	if err != nil {
+		zap.L().Error("failed to process template: %s", zap.Error(err))
```

Comment:
should we return 500 here?

Reply:
also. Should we update the state of the Token here?

Reply:
imo it's better to return 500 + at least the textual representation of the error if we cannot parse template